From a19bdb2b5e39e6d800bda77200d6b46880795b84 Mon Sep 17 00:00:00 2001
From: Ryan Lovelett <ryan@lovelett.me>
Date: Mon, 28 Dec 2015 15:34:30 -0500
Subject: [PATCH 2/6] [gyb] Python 2 and 3 compatible StringIO import

The `StringIO` and `cStringIO` modules are gone in Python 3. The
recommendation is to import the `io` module on Python 3. Therefore, this
patch first attempts to import the Python 2 module, `cStringIO`, and if
that fails then attempts to import the Python 3 module, `io`.

**NOTE**: There are still other Python 3.x fixes necessary to make `gyb`
run on a Python 3.x interpreter. This is just one small incremental
patch on the way there.
---
 utils/gyb.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/utils/gyb.py b/utils/gyb.py
index 4e98ced..eb17dd0 100755
--- a/utils/gyb.py
+++ b/utils/gyb.py
@@ -5,7 +5,10 @@
 from __future__ import print_function
 
 import re
-from cStringIO import StringIO
+try:
+    from cStringIO import StringIO
+except ImportError:
+    from io import StringIO
 import tokenize
 import textwrap
 from bisect import bisect
-- 
2.6.4

